[Win32] Change in ccflags between 5.39.6 and 5.39.7 is unacceptable.

**Description**
<!-- A clear and concise description of what the bug is. -->
On perl-5.39.6:
```
perl -V:ccflags
ccflags=' -DWIN32 -DWIN64 -fdiagnostics-color=never -DPERL_TEXTMODE_SCRIPTS -DMULTIPLICITY
 -DPERL_IMPLICIT_SYS -DUSE_PERLIO -D__USE_MINGW_ANSI_STDIO -fwrapv -fno-strict-aliasing -mms-bitfields';
```
On perl-5.39.7:
```
perl -V:ccflags
ccflags='-Wall -Wextra -Werror=pointer-arith -Wno-format -Wno-long-long -Werror=vla -std=c99 -DWIN32
 -DWIN64 -fdiagnostics-color=never -DPERL_TEXTMODE_SCRIPTS -DMULTIPLICITY
 -DPERL_IMPLICIT_SYS -DUSE_PERLIO -D__USE_MINGW_ANSI_STDIO -fwrapv -fno-strict-aliasing -mms-bitfields';
```
If that's deemed to not be a bug, then I have 2  observations to make:
1) I'll be patching my builds of perl to  return ccflags to the 5.39.6 rendition;
2) I'll be insisting that the same assault be launched on <b>all</b> systems, not just on Windows.

BTW, for anyone wondering where it comes from, the <code>-fdiagnostics-color=never</code> flag is a custom inclusion of mine.

For the record, both of those windows perls were built with the same configuration specs, and by the same mingw-w64 port of gcc-13.2.0.
On Ubuntu, for perl-5.39.7, I get:
```
$ perl -V:ccflags
ccflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong
 -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
```

> This was changed in https://github.com/Perl/perl5/commit/527e765cf579b4374cbeb60151f3d7bbb138bdf1.

Yes, but I thought that was going to apply only to the building of perl itself - which is horrifying enough. (Perl-5.39.7 now emits over 900 warnings during the build.)
I wasn't expecting that these same flags were going to be passed on to $Config{ccflags}. 

Consider a perl module that does
```
use overload  '+' => \&overload_add;
```
where overload_add() is an XSub.
Even though the third (switch) argument will never be used, perl still sends it anyway.
So the XSub has to allow for the fact that an unused argument is heading its way.
Usual thing is to therefore specify it in the argument list but do nothing with it.
But, of course, that triggers the <code>[-Wunused-parameter]</code> - as does every other "overload" XSub that's going to receive a third arg that it's never going to use.

I don't know how best to deal with that type of thing, off the top of my head.
Maybe make overload_add() a perl sub that passes only the 2 relevant args on to the XSub ?
Maybe use  the ellipsis syntax ?
I guess that if it's so important that unused parameters should not be sent, then the correct thing to do is to make the overload pragma stop sending that third arg when it's not going to be examined.
But there has been plenty of time for that to have been implemented already ... so I guess it's not that important after all ... which makes me wonder why we want to suddenly warning about it in the first place.

Thing is, I could spend a a month or two going through the tedium of fixing it in all my modules, and all I end up with is something that's no more efficient, and probably even less readable.
And all just to quiet the useless but distracting warning.
No thanks - I'd prefer to just remove the warnings in the first place. 

I find that some of the <code>[-Wunused-parameter]</code> warnings that I see during the build of perl itself are quite baffling.
For example, this one:
```
..\doio.c:64:24: warning: unused parameter 'fd' [-Wunused-parameter]
   64 | Perl_setfd_cloexec(int fd)
      | 
```
This complaint is raised in relation to:
```
void
Perl_setfd_cloexec(int fd)
{
    assert(fd >= 0);
#if defined(HAS_FCNTL) && defined(F_SETFD) && defined(FD_CLOEXEC)
    (void) fcntl(fd, F_SETFD, FD_CLOEXEC);
#endif
}
```
It looks to me that 'fd' <b>is</b> used. I don't know why the assert() call apparently doesn't count.
(There's other such warnings for similar reasons, too.)

Of course, there are other annoying warnings apart from the <code>[-Wunused-parameter]</code>, too.
I haven't looked at them closely yet, so I don't really know how tolerable I will find them to be.
But I haven't seen anything that looks particularly useful yet.

Anyway, if people think this should be done, then get it out there on linux, too.
Why should they be denied the joy that it will bring ?
For as long as it's just Windows that's afflicted, it  will appear to be an act of disingenuousness.

> I'd prefer such warnings were passed through to ccflags on Configure based systems too.

Go for it.
Until that happens, this will simply appear to be a vendetta against mingw-built windows.

There are some fairly significant differences between the warnings I see on msvc-built and mingw-built perl-5.39.7.

Firstly, 'cl' gives you just a one-line warning, whereas 'gcc' likes to give the warning, then write  a story about it.
 
Secondly, when building perl-5.39.7, of the 1551 warnings that 'cl' emits, 1345 are "possible loss of data", and a further 172 are "signed/unsigned mismatch".
In the remaining 34 warnings, there is no mention of anything being unused.

By contrast, the gcc build of the same perl source produces 35 [-Wunused- ...] warnings.
There are other differences as well - eg 'cl' compiles fcrypt.c without emitting any warnings, but 'gcc' produces:
```
fcrypt.c: In function 'des_fcrypt':
fcrypt.c:473:17: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
  473 |     ((c) >= '.' && (c) <= '9' || (c) >= 'A' && (c) <= 'Z' ||  \
      |      ~~~~~~~~~~~^~~~~~~~~~~~~
fcrypt.c:488:14: note: in expansion of macro 'good_for_salt'
  488 |         if (!good_for_salt(salt[0]) || !good_for_salt(salt[1])) {
      |              ^~~~~~~~~~~~~
fcrypt.c:474:17: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
  474 |      (c) >= 'a' && (c) <= 'z')
      |      ~~~~~~~~~~~^~~~~~~~~~~~~
fcrypt.c:488:14: note: in expansion of macro 'good_for_salt'
  488 |         if (!good_for_salt(salt[0]) || !good_for_salt(salt[1])) {
      |              ^~~~~~~~~~~~~
fcrypt.c:473:17: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
  473 |     ((c) >= '.' && (c) <= '9' || (c) >= 'A' && (c) <= 'Z' ||  \
      |      ~~~~~~~~~~~^~~~~~~~~~~~~
fcrypt.c:488:41: note: in expansion of macro 'good_for_salt'
  488 |         if (!good_for_salt(salt[0]) || !good_for_salt(salt[1])) {
      |                                         ^~~~~~~~~~~~~
fcrypt.c:474:17: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
  474 |      (c) >= 'a' && (c) <= 'z')
      |      ~~~~~~~~~~~^~~~~~~~~~~~~
fcrypt.c:488:41: note: in expansion of macro 'good_for_salt'
  488 |         if (!good_for_salt(salt[0]) || !good_for_salt(salt[1])) {
      |  
```

When these gcc flags are added to $Config{ccflags} things get quite nasty for me. On msvc-built 5.39.7,Math-MPFR-4.28 emits the following warnings:
```
MPFR.xs(254): warning C4018: '<': signed/unsigned mismatch
MPFR.xs(3114): warning C4018: '<': signed/unsigned mismatch
MPFR.xs(6177): warning C4244: 'function': conversion from 'UV' to 'unsigned long', possible loss of data
MPFR.xs(6178): warning C4244: 'function': conversion from 'UV' to 'long', possible loss of data
MPFR.xs(6169): warning C4101: 't': unreferenced local variable
MPFR.xs(6193): warning C4244: 'function': conversion from 'UV' to 'unsigned long', possible loss of data
MPFR.xs(6194): warning C4244: 'function': conversion from 'UV' to 'long', possible loss of data
MPFR.xs(6185): warning C4101: 't': unreferenced local variable
MPFR.xs(6203): warning C4244: 'function': conversion from 'UV' to 'unsigned long', possible loss of data
MPFR.xs(6204): warning C4244: 'function': conversion from 'UV' to 'long', possible loss of data
MPFR.xs(6214): warning C4244: 'function': conversion from 'UV' to 'unsigned long', possible loss of data
MPFR.xs(6215): warning C4244: 'function': conversion from 'UV' to 'long', possible loss of data
MPFR.xs(7894): warning C4018: '<': signed/unsigned mismatch
MPFR.xs(7899): warning C4018: '<': signed/unsigned mismatch
``` 
but on mingw-built 5.39.7 I get:
```
MPFR.xs: In function '_win32_infnanstring':
MPFR.xs:46:32: warning: unused parameter 's' [-Wunused-parameter]
   46 | int _win32_infnanstring(char * s) { /* MS Windows only - detect 1.#INF and 1.#IND
      |                         ~~~~~~~^
MPFR.xs: In function 'Rmpfr_clears':
MPFR.xs:254:19: warning: comparison of integer expressions of different signedness: 'long unsigned int' and 'Stack_off_t' {aka 'long int'} [-Wsign-compare]
  254 |      for(i = 0; i < items; i++) {
      |                   ^
MPFR.xs:251:30: warning: unused parameter 'p' [-Wunused-parameter]
  251 | void Rmpfr_clears(pTHX_ SV * p, ...) {
      |                         ~~~~~^
In file included from MPFR.xs:12:
MPFR.xs: In function 'Rmpfr_init_set':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:306:6: note: in expansion of macro 'dXSARGS'
  306 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_ui':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:321:6: note: in expansion of macro 'dXSARGS'
  321 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_si':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:336:6: note: in expansion of macro 'dXSARGS'
  336 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_d':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:351:6: note: in expansion of macro 'dXSARGS'
  351 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_ld':
MPFR.xs:365:35: warning: unused parameter 'q' [-Wunused-parameter]
  365 | void Rmpfr_init_set_ld(pTHX_ SV * q, SV * round) {
      |                              ~~~~~^
MPFR.xs:365:43: warning: unused parameter 'round' [-Wunused-parameter]
  365 | void Rmpfr_init_set_ld(pTHX_ SV * q, SV * round) {
      |                                      ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_init_set_f':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:384:6: note: in expansion of macro 'dXSARGS'
  384 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_z':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:399:6: note: in expansion of macro 'dXSARGS'
  399 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_q':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:414:6: note: in expansion of macro 'dXSARGS'
  414 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_str':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:429:6: note: in expansion of macro 'dXSARGS'
  429 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:465:6: note: in expansion of macro 'dXSARGS'
  465 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_ui_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:480:6: note: in expansion of macro 'dXSARGS'
  480 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_si_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:495:6: note: in expansion of macro 'dXSARGS'
  495 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_d_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:510:6: note: in expansion of macro 'dXSARGS'
  510 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_ld_nobless':
MPFR.xs:524:43: warning: unused parameter 'q' [-Wunused-parameter]
  524 | void Rmpfr_init_set_ld_nobless(pTHX_ SV * q, SV * round) {
      |                                      ~~~~~^
MPFR.xs:524:51: warning: unused parameter 'round' [-Wunused-parameter]
  524 | void Rmpfr_init_set_ld_nobless(pTHX_ SV * q, SV * round) {
      |                                              ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_init_set_f_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:545:6: note: in expansion of macro 'dXSARGS'
  545 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_z_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:560:6: note: in expansion of macro 'dXSARGS'
  560 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_q_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:575:6: note: in expansion of macro 'dXSARGS'
  575 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_str_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:590:6: note: in expansion of macro 'dXSARGS'
  590 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_deref2':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:609:6: note: in expansion of macro 'dXSARGS'
  609 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_NV':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:775:6: note: in expansion of macro 'dXSARGS'
  775 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_NV_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:792:6: note: in expansion of macro 'dXSARGS'
  792 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_cmp_float128':
MPFR.xs:808:39: warning: unused parameter 'a' [-Wunused-parameter]
  808 | int Rmpfr_cmp_float128(pTHX_ mpfr_t * a, SV * b) {
      |                              ~~~~~~~~~^
MPFR.xs:808:47: warning: unused parameter 'b' [-Wunused-parameter]
  808 | int Rmpfr_cmp_float128(pTHX_ mpfr_t * a, SV * b) {
      |                                          ~~~~~^
MPFR.xs: In function 'Rmpfr_set_ld':
MPFR.xs:901:34: warning: unused parameter 'p' [-Wunused-parameter]
  901 | SV * Rmpfr_set_ld(pTHX_ mpfr_t * p, SV * q, SV * round) {
      |                         ~~~~~~~~~^
MPFR.xs:901:42: warning: unused parameter 'q' [-Wunused-parameter]
  901 | SV * Rmpfr_set_ld(pTHX_ mpfr_t * p, SV * q, SV * round) {
      |                                     ~~~~~^
MPFR.xs:901:50: warning: unused parameter 'round' [-Wunused-parameter]
  901 | SV * Rmpfr_set_ld(pTHX_ mpfr_t * p, SV * q, SV * round) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_get_ld_2exp':
MPFR.xs:998:35: warning: unused parameter 'exp' [-Wunused-parameter]
  998 | SV * Rmpfr_get_ld_2exp(pTHX_ SV * exp, mpfr_t * p, SV * round){
      |                              ~~~~~^~~
MPFR.xs:998:49: warning: unused parameter 'p' [-Wunused-parameter]
  998 | SV * Rmpfr_get_ld_2exp(pTHX_ SV * exp, mpfr_t * p, SV * round){
      |                                        ~~~~~~~~~^
MPFR.xs:998:57: warning: unused parameter 'round' [-Wunused-parameter]
  998 | SV * Rmpfr_get_ld_2exp(pTHX_ SV * exp, mpfr_t * p, SV * round){
      |                                                    ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_get_ld':
MPFR.xs:1028:34: warning: unused parameter 'p' [-Wunused-parameter]
 1028 | SV * Rmpfr_get_ld(pTHX_ mpfr_t * p, SV * round){
      |                         ~~~~~~~~~^
MPFR.xs:1028:42: warning: unused parameter 'round' [-Wunused-parameter]
 1028 | SV * Rmpfr_get_ld(pTHX_ mpfr_t * p, SV * round){
      |                                     ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_cmp_ld':
MPFR.xs:1564:33: warning: unused parameter 'a' [-Wunused-parameter]
 1564 | int Rmpfr_cmp_ld(pTHX_ mpfr_t * a, SV * b) {
      |                        ~~~~~~~~~^
MPFR.xs:1564:41: warning: unused parameter 'b' [-Wunused-parameter]
 1564 | int Rmpfr_cmp_ld(pTHX_ mpfr_t * a, SV * b) {
      |                                    ~~~~~^
MPFR.xs: In function 'Rmpfr_urandomb':
MPFR.xs:2022:32: warning: unused parameter 'x' [-Wunused-parameter]
 2022 | void Rmpfr_urandomb(pTHX_ SV * x, ...) {
      |                           ~~~~~^
MPFR.xs: In function 'Rmpfr_random2':
MPFR.xs:2035:35: warning: unused parameter 'p' [-Wunused-parameter]
 2035 | void Rmpfr_random2(pTHX_ mpfr_t * p, SV * s, SV * exp) {
      |                          ~~~~~~~~~^
MPFR.xs:2035:43: warning: unused parameter 's' [-Wunused-parameter]
 2035 | void Rmpfr_random2(pTHX_ mpfr_t * p, SV * s, SV * exp) {
      |                                      ~~~~~^
MPFR.xs:2035:51: warning: unused parameter 'exp' [-Wunused-parameter]
 2035 | void Rmpfr_random2(pTHX_ mpfr_t * p, SV * s, SV * exp) {
      |                                              ~~~~~^~~
MPFR.xs: In function 'Rmpfr_remquo':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:2251:6: note: in expansion of macro 'dXSARGS'
 2251 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_IV':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:2592:6: note: in expansion of macro 'dXSARGS'
 2592 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_init_set_IV_nobless':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:2609:6: note: in expansion of macro 'dXSARGS'
 2609 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_root':
MPFR.xs:3012:32: warning: unused parameter 'a' [-Wunused-parameter]
 3012 | SV * Rmpfr_root(pTHX_ mpfr_t * a, mpfr_t * b, SV * c, SV * round) {
      |                       ~~~~~~~~~^
MPFR.xs:3012:44: warning: unused parameter 'b' [-Wunused-parameter]
 3012 | SV * Rmpfr_root(pTHX_ mpfr_t * a, mpfr_t * b, SV * c, SV * round) {
      |                                   ~~~~~~~~~^
MPFR.xs:3012:52: warning: unused parameter 'c' [-Wunused-parameter]
 3012 | SV * Rmpfr_root(pTHX_ mpfr_t * a, mpfr_t * b, SV * c, SV * round) {
      |                                               ~~~~~^
MPFR.xs:3012:60: warning: unused parameter 'round' [-Wunused-parameter]
 3012 | SV * Rmpfr_root(pTHX_ mpfr_t * a, mpfr_t * b, SV * c, SV * round) {
      |                                                       ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_lgamma':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:3064:6: note: in expansion of macro 'dXSARGS'
 3064 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_sum':
MPFR.xs:3114:19: warning: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [-Wsign-compare]
 3114 |      for(i = 0; i < s; ++i) {
      |                   ^
MPFR.xs: In function 'overload_mul':
MPFR.xs:3194:46: warning: unused parameter 'third' [-Wunused-parameter]
 3194 | SV * overload_mul(pTHX_ SV * a, SV * b, SV * third) {
      |                                         ~~~~~^~~~~
MPFR.xs: In function 'overload_add':
MPFR.xs:3328:46: warning: unused parameter 'third' [-Wunused-parameter]
 3328 | SV * overload_add(pTHX_ SV * a, SV * b, SV * third) {
      |                                         ~~~~~^~~~~
MPFR.xs: In function 'overload_copy':
MPFR.xs:3769:43: warning: unused parameter 'b' [-Wunused-parameter]
 3769 | SV * overload_copy(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                      ~~~~~^
MPFR.xs:3769:51: warning: unused parameter 'third' [-Wunused-parameter]
 3769 | SV * overload_copy(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                              ~~~~~^~~~~
MPFR.xs: In function 'overload_abs':
MPFR.xs:3781:42: warning: unused parameter 'b' [-Wunused-parameter]
 3781 | SV * overload_abs(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:3781:50: warning: unused parameter 'third' [-Wunused-parameter]
 3781 | SV * overload_abs(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_equiv':
MPFR.xs:4340:52: warning: unused parameter 'third' [-Wunused-parameter]
 4340 | SV * overload_equiv(pTHX_ mpfr_t * a, SV * b, SV * third) {
      |                                               ~~~~~^~~~~
MPFR.xs: In function 'overload_not_equiv':
MPFR.xs:4444:56: warning: unused parameter 'third' [-Wunused-parameter]
 4444 | SV * overload_not_equiv(pTHX_ mpfr_t * a, SV * b, SV * third) {
      |                                                   ~~~~~^~~~~
MPFR.xs: In function 'overload_true':
MPFR.xs:4549:41: warning: unused parameter 'b' [-Wunused-parameter]
 4549 | SV * overload_true(pTHX_ mpfr_t *a, SV *b, SV * third) {
      |                                     ~~~~^
MPFR.xs:4549:49: warning: unused parameter 'third' [-Wunused-parameter]
 4549 | SV * overload_true(pTHX_ mpfr_t *a, SV *b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'overload_not':
MPFR.xs:4555:42: warning: unused parameter 'b' [-Wunused-parameter]
 4555 | SV * overload_not(pTHX_ mpfr_t * a, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4555:50: warning: unused parameter 'third' [-Wunused-parameter]
 4555 | SV * overload_not(pTHX_ mpfr_t * a, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_sqrt':
MPFR.xs:4561:43: warning: unused parameter 'b' [-Wunused-parameter]
 4561 | SV * overload_sqrt(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                      ~~~~~^
MPFR.xs:4561:51: warning: unused parameter 'third' [-Wunused-parameter]
 4561 | SV * overload_sqrt(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                              ~~~~~^~~~~
MPFR.xs: In function 'overload_log':
MPFR.xs:4724:42: warning: unused parameter 'b' [-Wunused-parameter]
 4724 | SV * overload_log(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4724:50: warning: unused parameter 'third' [-Wunused-parameter]
 4724 | SV * overload_log(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_exp':
MPFR.xs:4736:42: warning: unused parameter 'b' [-Wunused-parameter]
 4736 | SV * overload_exp(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4736:50: warning: unused parameter 'third' [-Wunused-parameter]
 4736 | SV * overload_exp(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_sin':
MPFR.xs:4748:42: warning: unused parameter 'b' [-Wunused-parameter]
 4748 | SV * overload_sin(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4748:50: warning: unused parameter 'third' [-Wunused-parameter]
 4748 | SV * overload_sin(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_cos':
MPFR.xs:4760:42: warning: unused parameter 'b' [-Wunused-parameter]
 4760 | SV * overload_cos(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4760:50: warning: unused parameter 'third' [-Wunused-parameter]
 4760 | SV * overload_cos(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_int':
MPFR.xs:4772:42: warning: unused parameter 'b' [-Wunused-parameter]
 4772 | SV * overload_int(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                     ~~~~~^
MPFR.xs:4772:50: warning: unused parameter 'third' [-Wunused-parameter]
 4772 | SV * overload_int(pTHX_ mpfr_t * p, SV * b, SV * third) {
      |                                             ~~~~~^~~~~
MPFR.xs: In function 'overload_pow_eq':
MPFR.xs:5036:49: warning: unused parameter 'third' [-Wunused-parameter]
 5036 | SV * overload_pow_eq(pTHX_ SV * p, SV * b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'overload_div_eq':
MPFR.xs:5167:49: warning: unused parameter 'third' [-Wunused-parameter]
 5167 | SV * overload_div_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'overload_sub_eq':
MPFR.xs:5289:49: warning: unused parameter 'third' [-Wunused-parameter]
 5289 | SV * overload_sub_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'overload_add_eq':
MPFR.xs:5421:49: warning: unused parameter 'third' [-Wunused-parameter]
 5421 | SV * overload_add_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'overload_mul_eq':
MPFR.xs:5545:49: warning: unused parameter 'third' [-Wunused-parameter]
 5545 | SV * overload_mul_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_grandom':
MPFR.xs:6112:35: warning: unused parameter 'rop1' [-Wunused-parameter]
 6112 | SV * Rmpfr_grandom(pTHX_ mpfr_t * rop1, mpfr_t * rop2, gmp_randstate_t * state, SV * round) {
      |                          ~~~~~~~~~^~~~
MPFR.xs:6112:50: warning: unused parameter 'rop2' [-Wunused-parameter]
 6112 | SV * Rmpfr_grandom(pTHX_ mpfr_t * rop1, mpfr_t * rop2, gmp_randstate_t * state, SV * round) {
      |                                         ~~~~~~~~~^~~~
MPFR.xs:6112:74: warning: unused parameter 'state' [-Wunused-parameter]
 6112 | SV * Rmpfr_grandom(pTHX_ mpfr_t * rop1, mpfr_t * rop2, gmp_randstate_t * state, SV * round) {
      |                                                        ~~~~~~~~~~~~~~~~~~^~~~~
MPFR.xs:6112:86: warning: unused parameter 'round' [-Wunused-parameter]
 6112 | SV * Rmpfr_grandom(pTHX_ mpfr_t * rop1, mpfr_t * rop2, gmp_randstate_t * state, SV * round) {
      |                                                                                 ~~~~~^~~~~
MPFR.xs: In function 'overload_inc':
MPFR.xs:6158:38: warning: unused parameter 'b' [-Wunused-parameter]
 6158 | void overload_inc(pTHX_ SV * a, SV * b, SV * third) {
      |                                 ~~~~~^
MPFR.xs:6158:46: warning: unused parameter 'third' [-Wunused-parameter]
 6158 | void overload_inc(pTHX_ SV * a, SV * b, SV * third) {
      |                                         ~~~~~^~~~~
MPFR.xs: In function 'overload_dec':
MPFR.xs:6163:38: warning: unused parameter 'b' [-Wunused-parameter]
 6163 | void overload_dec(pTHX_ SV * a, SV * b, SV * third) {
      |                                 ~~~~~^
MPFR.xs:6163:46: warning: unused parameter 'third' [-Wunused-parameter]
 6163 | void overload_dec(pTHX_ SV * a, SV * b, SV * third) {
      |                                         ~~~~~^~~~~
MPFR.xs: In function 'overload_mul_2exp':
MPFR.xs:6169:27: warning: unused variable 't' [-Wunused-variable]
 6169 |      mpfr_t * mpfr_t_obj, t;
      |                           ^
MPFR.xs:6168:51: warning: unused parameter 'third' [-Wunused-parameter]
 6168 | SV * overload_mul_2exp(pTHX_ SV * a, SV * b, SV * third) {
      |                                              ~~~~~^~~~~
MPFR.xs: In function 'overload_div_2exp':
MPFR.xs:6185:27: warning: unused variable 't' [-Wunused-variable]
 6185 |      mpfr_t * mpfr_t_obj, t;
      |                           ^
MPFR.xs:6184:51: warning: unused parameter 'third' [-Wunused-parameter]
 6184 | SV * overload_div_2exp(pTHX_ SV * a, SV * b, SV * third) {
      |                                              ~~~~~^~~~~
MPFR.xs: In function 'overload_mul_2exp_eq':
MPFR.xs:6200:54: warning: unused parameter 'third' [-Wunused-parameter]
 6200 | SV * overload_mul_2exp_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                                 ~~~~~^~~~~
MPFR.xs: In function 'overload_div_2exp_eq':
MPFR.xs:6211:54: warning: unused parameter 'third' [-Wunused-parameter]
 6211 | SV * overload_div_2exp_eq(pTHX_ SV * a, SV * b, SV * third) {
      |                                                 ~~~~~^~~~~
MPFR.xs: In function '_mp_sizes':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:6468:6: note: in expansion of macro 'dXSARGS'
 6468 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_get_float128':
MPFR.xs:6541:40: warning: unused parameter 'op' [-Wunused-parameter]
 6541 | SV * Rmpfr_get_float128(pTHX_ mpfr_t * op, SV * rnd) {
      |                               ~~~~~~~~~^~
MPFR.xs:6541:49: warning: unused parameter 'rnd' [-Wunused-parameter]
 6541 | SV * Rmpfr_get_float128(pTHX_ mpfr_t * op, SV * rnd) {
      |                                            ~~~~~^~~
MPFR.xs: In function 'Rmpfr_set_float128':
MPFR.xs:6625:40: warning: unused parameter 'rop' [-Wunused-parameter]
 6625 | SV * Rmpfr_set_float128(pTHX_ mpfr_t * rop, SV * q, SV * rnd) {
      |                               ~~~~~~~~~^~~
MPFR.xs:6625:50: warning: unused parameter 'q' [-Wunused-parameter]
 6625 | SV * Rmpfr_set_float128(pTHX_ mpfr_t * rop, SV * q, SV * rnd) {
      |                                             ~~~~~^
MPFR.xs:6625:58: warning: unused parameter 'rnd' [-Wunused-parameter]
 6625 | SV * Rmpfr_set_float128(pTHX_ mpfr_t * rop, SV * q, SV * rnd) {
      |                                                     ~~~~~^~~
MPFR.xs: In function 'Rmpfr_init_set_float128':
MPFR.xs:6642:41: warning: unused parameter 'q' [-Wunused-parameter]
 6642 | void Rmpfr_init_set_float128(pTHX_ SV * q, SV * round) {
      |                                    ~~~~~^
MPFR.xs:6642:49: warning: unused parameter 'round' [-Wunused-parameter]
 6642 | void Rmpfr_init_set_float128(pTHX_ SV * q, SV * round) {
      |                                            ~~~~~^~~~~
MPFR.xs: In function 'Rmpfr_init_set_float128_nobless':
MPFR.xs:6666:49: warning: unused parameter 'q' [-Wunused-parameter]
 6666 | void Rmpfr_init_set_float128_nobless(pTHX_ SV * q, SV * round) {
      |                                            ~~~~~^
MPFR.xs:6666:57: warning: unused parameter 'round' [-Wunused-parameter]
 6666 | void Rmpfr_init_set_float128_nobless(pTHX_ SV * q, SV * round) {
      |                                                    ~~~~~^~~~~
MPFR.xs: In function '_bytes_fr':
MPFR.xs:6861:26: warning: comparison of integer expressions of different signedness: 'mpfr_prec_t' {aka 'long int'} and 'unsigned int' [-Wsign-compare]
 6861 |   if(mpfr_get_prec(*str) != bits)
      |                          ^~
MPFR.xs: In function 'Rmpfr_fmodquo':
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:166:28: warning: unused variable 'items' [-Wunused-variable]
  166 | #define dITEMS Stack_off_t items = (Stack_off_t)(SP - MARK)
      |                            ^~~~~
D:\perl-5.39.7-1320-ucrt-mcf\lib\MSWin32-x64-multi-thread\CORE/XSUB.h:169:23: note: in expansion of macro 'dITEMS'
  169 |         dSP; dAXMARK; dITEMS
      |                       ^~~~~~
MPFR.xs:7258:6: note: in expansion of macro 'dXSARGS'
 7258 |      dXSARGS;
      |      ^~~~~~~
MPFR.xs: In function 'Rmpfr_dot':
MPFR.xs:7894:19: warning: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [-Wsign-compare]
 7894 |      for(i = 0; i < s; ++i) {
      |                   ^
MPFR.xs:7899:19: warning: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [-Wsign-compare]
 7899 |      for(i = 0; i < s; ++i) {
      |                   ^
MPFR.xs: In function '_nvtoa':
MPFR.xs:8056:8: warning: unused variable 'str' [-Wunused-variable]
 8056 |   char str[] = {'\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0'};
      |        ^~~
MPFR.xs: In function '_mpfrtoa':
MPFR.xs:8470:8: warning: unused variable 'str' [-Wunused-variable]
 8470 |   char str[] = {'\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0','\0'};
      |        ^~~
MPFR.xs: In function 'decimalize':
MPFR.xs:8933:5: warning: 'low_exp' may be used uninitialized [-Wmaybe-uninitialized]
 8933 |   if(low_exp >= 0) {
      |     ^
MPFR.xs:8868:29: note: 'low_exp' was declared here
 8868 |   mpfr_exp_t exp, high_exp, low_exp;
      |                             ^~~~~~~
```
I have, of course, silenced those warnings in the github repo version of the module, but haven't yet bothered with a new release. (I'm probably the only person that uses that module, anyway.)
However, it seems to me that the 2 warnings regimes (ie 'cl' versus 'gcc') are more different in 5.39.7 than they were in 5.39.6.

Still ... I can always set the warnings flags to whatever I want when I build perl, so I've probably got nothing to complain about.
I'm not sure how the Strawberry Perl developers will take to this new innovation. I guess that @genio and @shawnlaffan will just deal with it as they see fit.

--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -619,7 +619,7 @@ OPTIMIZE	= -Os
 LINK_DBG	= -s
 endif
 
-EXTRACFLAGS	= -Wall -Wextra -Werror=pointer-arith -Wno-format -Wno-long-long -Werror=vla
+CWARNFLAGS	= -Wall -Wextra -Werror=pointer-arith -Wno-format -Wno-long-long -Werror=vla
 ifeq ($(USE_CPLUSPLUS),define)
 EXTRACFLAGS	+= $(CXX_FLAG)
 else
@@ -763,7 +763,7 @@ TESTPREPGCC	=
 
 endif
 
-CFLAGS_O	= $(CFLAGS) $(BUILDOPT)
+CFLAGS_O	= $(CFLAGS) $(BUILDOPT) $(CWARNFLAGS)
 
 RSC_FLAGS	=
 
@@ -1424,10 +1424,10 @@ endif
 	@rem. > $(MINIDIR)\.exists
 
 $(MINIDIR)\\%$(o): %.c
-	$(CC) -c -I$(MINIDIR) $(CFLAGS) $(MINIBUILDOPT) -DPERL_EXTERNAL_GLOB -DPERL_IS_MINIPERL $(OBJOUT_FLAG)$@ $(PDBOUT) $<
+	$(CC) -c -I$(MINIDIR) $(CFLAGS) $(CWARNFLAGS) $(MINIBUILDOPT) -DPERL_EXTERNAL_GLOB -DPERL_IS_MINIPERL $(OBJOUT_FLAG)$@ $(PDBOUT) $<
 
 $(MINIDIR)\\%$(o): ..\%.c
-	$(CC) -c -I$(MINIDIR) $(CFLAGS) $(MINIBUILDOPT) -DPERL_EXTERNAL_GLOB -DPERL_IS_MINIPERL $(OBJOUT_FLAG)$@ $(PDBOUT) $<
+	$(CC) -c -I$(MINIDIR) $(CFLAGS) $(CWARNFLAGS) $(MINIBUILDOPT) -DPERL_EXTERNAL_GLOB -DPERL_IS_MINIPERL $(OBJOUT_FLAG)$@ $(PDBOUT) $<
 
 $(MINICORE_OBJ) : $(CORE_NOCFG_H)
 

